From c0bd0c65c216796321650a58bd0fd505ef23406b Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 28 Mar 2016 21:22:26 -0400
Subject: [PATCH] EntityPathfindEvent

Fires when an Entity decides to start moving to a location.

diff --git a/src/main/java/net/minecraft/server/NavigationAbstract.java b/src/main/java/net/minecraft/server/NavigationAbstract.java
index cc44d30b..3ac6f84d 100644
--- a/src/main/java/net/minecraft/server/NavigationAbstract.java
+++ b/src/main/java/net/minecraft/server/NavigationAbstract.java
@@ -4,7 +4,7 @@ import javax.annotation.Nullable;
 
 public abstract class NavigationAbstract {
 
-    protected EntityInsentient a;
+    protected EntityInsentient a; public Entity getEntity() { return a; } // Paper - OBFHELPER
     protected World b;
     @Nullable
     protected PathEntity c;
@@ -74,6 +74,7 @@ public abstract class NavigationAbstract {
         } else if (this.c != null && !this.c.b() && blockposition.equals(this.q)) {
             return this.c;
         } else {
+            if (!new com.destroystokyo.paper.event.entity.EntityPathfindEvent(getEntity().getBukkitEntity(), MCUtil.toLocation(getEntity().world, blockposition), null).callEvent()) { return null; } // Paper
             this.q = blockposition;
             float f = this.i();
 
@@ -98,6 +99,7 @@ public abstract class NavigationAbstract {
             if (this.c != null && !this.c.b() && blockposition.equals(this.q)) {
                 return this.c;
             } else {
+                if (!new com.destroystokyo.paper.event.entity.EntityPathfindEvent(getEntity().getBukkitEntity(), MCUtil.toLocation(entity.world, blockposition), entity.getBukkitEntity()).callEvent()) { return null; } // Paper
                 this.q = blockposition;
                 float f = this.i();
 
-- 
2.14.3

